src/libsyntax_ext/deriving/generic/mod.rs

```diff
@@ -598,7 +598,7 @@ impl<'a> TraitDef<'a> {
 
                     let predicate = ast::WhereBoundPredicate {
                         span: self.span,
-                        bound_lifetimes: vec![],
+                        bound_generics: Default::default(),
```

Comment:
Here, and in a few other places, you have `Default::default()`, I would prefer `Generics::default()` for clarity.